test/unit/math/test_ad.hpp

```diff
@@ -1432,6 +1433,54 @@ void expect_ad_vectorized_binary(const F& f, const T1& x, const T2& y) {
   expect_ad_vectorized_binary(tols, f, x, y);
 }
 
+template <typename T1, typename T2,
+          require_all_stan_scalar_t<T1, T2>* = nullptr>
+void expect_float_equal(const T1& x, const T2& y) {
+  EXPECT_FLOAT_EQ(x.val(), y.val());
```

Comment:
This needs to be more than `EXPECT_FLOAT_EQ`. Check here: https://github.com/stan-dev/math/blob/develop/test/unit/math/test_ad.hpp#L122

Reply:
Still needs changed to use `expect_near_rel` or removed. `expect_near_rel` is set up to handle values near zero better than `EXPECT_FLOAT_EQ`. If `EXPECT_FLOAT_EQ` worked, then `expect_near_rel` probably will. It's a better way of doing these comparisons so edge cases don't trip you up.